iot-hub/Quickstarts/simulated-device-2/SimulatedDeviceSync.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) Microsoft. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import random
+import time
+import os
+from azure.iot.device import IoTHubDeviceClient, Message, MethodResponse
+
+# The device connection string to authenticate the device with your IoT hub.
+# Using the Azure CLI:
+# az iot hub device-identity show-connection-string --hub-name {YourIoTHubName} --device-id MyNodeDevice --output table
+CONNECTION_STRING = os.getenv("IOTHUB_DEVICE_CONNECTION_STRING")
+
+# Define the JSON message to send to IoT Hub.
+TEMPERATURE = 20.0
+HUMIDITY = 60
+MSG_TXT = '{{"temperature": {temperature},"humidity": {humidity}}}'
+
+INTERVAL = 1
+
+
+def create_client():
+    # Create an IoT Hub client
+    client = IoTHubDeviceClient.create_from_connection_string(CONNECTION_STRING)
+
+    # Define a method request handler
+    def method_request_handler(method_request):
+        if method_request.name == "SetTelemetryInterval":
+            try:
+                global INTERVAL
+                INTERVAL = int(method_request.payload)
+            except ValueError:
+                response_payload = {"Response": "Invalid parameter"}
+                response_status = 400
+            else:
+                response_payload = {"Response": "Executed direct method {}".format(method_request.name)}
+                response_status = 200
+        else:
+            response_payload = {"Response": "Direct method {} not defined".format(method_request.name)}
+            response_status = 404
+
+        method_response = MethodResponse.create_from_method_request(method_request, response_status, response_payload)
+        client.send_method_response(method_response)
+
+    # Attach the method request handler
+    client.on_method_request_received = method_request_handler
```

Comment:
Technically there could be a failure while attaching handlers, and if you were setting multiple handlers and, say, the second one failed, you would ideally do a graceful shutdown using the .shutdown() method here. If just attaching one handler, it's not really necessary, but IDK, it might be a good idea to show the best practice?